File: VB6Conv/VBA2VBProject.NET/frmVfcMain.vb
Option Strict Off
Option Explicit On
Friend Class frmVfcMain
	Inherits System.Windows.Forms.Form
	
	
	Public ReadOnly Property SelectedProjectIndex() As Integer
		Get
			If lstVBProjects.SelectedIndex > -1 Then
				SelectedProjectIndex = CInt(lstVBProjects.Text)
			End If
		End Get
	End Property
	
	Public ReadOnly Property SelectedProjectFilename() As String
		Get
			If lstVBProjects.SelectedIndex > -1 Then
				SelectedProjectFilename = lstVBProjects.List(lstVBProjects.SelectedIndex, 2)
			End If
		End Get
	End Property
	
	Public ReadOnly Property SelectedProjectGitDirectory() As String
		Get
			If lstVBProjects.SelectedIndex > -1 Then
                SelectedProjectGitDirectory = txtGitRepoPath.Text & StripExtensionFromFileName(GetFileNameFromPath(lstVBProjects.SelectedItem(lstVBProjects.SelectedIndex, 2)))
				'UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
				If Dir(SelectedProjectGitDirectory, FileAttribute.Directory) = "" Then MkDir(SelectedProjectGitDirectory)
			End If
		End Get
	End Property
	
	Public ReadOnly Property SelectedProjectGitIgnoreDirectory() As String
		Get
			If lstVBProjects.SelectedIndex > -1 Then
				SelectedProjectGitIgnoreDirectory = SelectedProjectGitDirectory & "\ignore\"
				'UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
				If Dir(SelectedProjectGitIgnoreDirectory, FileAttribute.Directory) = "" Then MkDir(SelectedProjectGitIgnoreDirectory)
			End If
		End Get
	End Property
	
	
	Public ReadOnly Property IncludeCode() As Boolean
		Get
			IncludeCode = (chkIncludeCode.CheckState = True)
		End Get
	End Property
	
	Public ReadOnly Property ShowUnknown() As Boolean
		Get
			ShowUnknown = (chkShowUnknown.CheckState = True)
		End Get
	End Property
	
	Private Sub cmdBrowseRepo_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdBrowseRepo.Click
		
		txtGitRepoPath.Text = BrowseFolder("Select GIT Repo Base Folder", Environ("USERPROFILE") & "\Source\Repos\Autocad Automation\")
		'MsgBox Environ("USERPROFILE")
		'"C:\Users\acunningham\Source\Repos"
	End Sub
	
	Private Sub cmdCancel_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdCancel.Click
		Me.Close()
	End Sub
	
	Private Sub cmdConvert_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdConvert.Click
		cmdCancel.Text = "Exit"
		Call ProcessProject()
		Call ExportSourceFiles(SelectedProjectGitDirectory & "\")
	End Sub
	
	
	
	'UPGRADE_WARNING: Event lstVBProjects.SelectedIndexChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
	Private Sub lstVBProjects_SelectedIndexChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles lstVBProjects.SelectedIndexChanged
		Dim objProj As Microsoft.Vbe.Interop.VBProject
		Dim intIndex As Short
		
		If lstVBProjects.SelectedIndex > -1 Then
			'UPGRADE_WARNING: Couldn't resolve default property of object Application.VBE.VBProjects. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
			objProj = AutoCADAcadApplication_definst.Application.VBE.VBProjects(Me.SelectedProjectIndex)
			cmdConvert.Enabled = True
			
		End If
		
		'UPGRADE_NOTE: Object objProj may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
		objProj = Nothing
	End Sub
	
	
	
	Private Sub UserForm_Activate()
		Dim objProjs As Microsoft.Vbe.Interop.VBProjects
		Dim objProj As Microsoft.Vbe.Interop.VBProject
		Dim intIndex As Short
		Dim strTemp As String
		
		On Error GoTo ErrorHandler
		
		'Load up list box with VBProjects
		'UPGRADE_WARNING: Couldn't resolve default property of object Application.VBE.VBProjects. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
		objProjs = AutoCADAcadApplication_definst.Application.VBE.VBProjects
		cmdConvert.Enabled = False
		'  lstMSForms.Clear
		lstVBProjects.Items.Clear()
		'UPGRADE_WARNING: Couldn't resolve default property of object lstVBProjects.ColumnCount. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
		lstVBProjects.ColumnCount = 3
		'UPGRADE_WARNING: Couldn't resolve default property of object lstVBProjects.ColumnWidths. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
		lstVBProjects.ColumnWidths = "20 pt;90 pt;400 pt"
		Dim listIndex As Short
		listIndex = 1
		For intIndex = 1 To objProjs.Count
			objProj = objProjs.Item(intIndex)
			If (objProj.fileName <> objProjs.VBE.ActiveVBProject.fileName) Then 'do not list the exporter macro
				lstVBProjects.Items.Add(CStr(intIndex))
				lstVBProjects.List(listIndex - 1, 1) = objProj.Name
				lstVBProjects.List(listIndex - 1, 2) = objProj.fileName
				listIndex = listIndex + 1
			End If
		Next intIndex
		txtGitRepoPath.Text = Environ("USERPROFILE") & "\Source\Repos\Autocad Automation\"
		
		
SubExit: 
		'UPGRADE_NOTE: Object objProj may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
		objProj = Nothing
		'UPGRADE_NOTE: Object objProjs may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
		objProjs = Nothing
		Exit Sub
ErrorHandler: 
		Select Case Err.Number
			Case 76 'Path not found
				strTemp = "VFC can only work with SAVED projects," & vbCrLf & "please save all newly created projects" & vbCrLf & "and start again."
				Call MsgBox(strTemp, MsgBoxStyle.Critical, ccAPPNAME)
			Case Else 'Huh??
				strTemp = ccAPPNAME & " ERROR" & vbCrLf & "VFC_001: Error in UserForm_Activate" & vbCrLf & "Description: " & Err.Description & vbCrLf & "Source: " & Err.Source & vbCrLf & "Number: " & CStr(Err.Number) & vbCrLf & "VFC Ver: " & ccAPPVER
				Call MsgBox(strTemp, MsgBoxStyle.Critical, ccAPPNAME)
		End Select
		Me.Close()
		Resume SubExit
	End Sub
	
	Private Sub UserForm_Terminate()
		Debug.Print("UserForm_Terminate frmVfcMain")
	End Sub
End Class

File: VB6Conv/VBA2VBProject.NET.ConvertedToC#/frmVfcMain.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.Compatibility;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;
 // ERROR: Not supported in C#: OptionDeclaration
namespace VBA2VBProject
{
	internal partial class frmVfcMain : System.Windows.Forms.Form
	{


		public int SelectedProjectIndex {
			get {
				int functionReturnValue = 0;
				if (lstVBProjects.SelectedIndex > -1) {
					functionReturnValue = Convert.ToInt32(lstVBProjects.Text);
				}
				return functionReturnValue;
			}
		}

		public string SelectedProjectFilename {
			get {
				string functionReturnValue = null;
				if (lstVBProjects.SelectedIndex > -1) {
					functionReturnValue = lstVBProjects.List(lstVBProjects.SelectedIndex, 2);
				}
				return functionReturnValue;
			}
		}

		public string SelectedProjectGitDirectory {
			get {
				string functionReturnValue = null;
				if (lstVBProjects.SelectedIndex > -1) {
					functionReturnValue = txtGitRepoPath.Text + modFileFunctions.StripExtensionFromFileName(modFileFunctions.GetFileNameFromPath(lstVBProjects.SelectedItem(lstVBProjects.SelectedIndex, 2)));
					//UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
					if (string.IsNullOrEmpty(FileSystem.Dir(SelectedProjectGitDirectory, FileAttribute.Directory)))
						FileSystem.MkDir(SelectedProjectGitDirectory);
				}
				return functionReturnValue;
			}
		}

		public string SelectedProjectGitIgnoreDirectory {
			get {
				string functionReturnValue = null;
				if (lstVBProjects.SelectedIndex > -1) {
					functionReturnValue = SelectedProjectGitDirectory + "\\ignore\\";
					//UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
					if (string.IsNullOrEmpty(FileSystem.Dir(SelectedProjectGitIgnoreDirectory, FileAttribute.Directory)))
						FileSystem.MkDir(SelectedProjectGitIgnoreDirectory);
				}
				return functionReturnValue;
			}
		}


		public bool IncludeCode {
			get { return (chkIncludeCode.CheckState == true); }
		}

		public bool ShowUnknown {
			get { return (chkShowUnknown.CheckState == true); }
		}

		private void cmdBrowseRepo_Click(System.Object eventSender, System.EventArgs eventArgs)
		{

			txtGitRepoPath.Text = modFileFunctions.BrowseFolder(ref "Select GIT Repo Base Folder", ref Interaction.Environ("USERPROFILE") + "\\Source\\Repos\\Autocad Automation\\");
			//MsgBox Environ("USERPROFILE")
			//"C:\Users\acunningham\Source\Repos"
		}

		private void cmdCancel_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			this.Close();
		}

		private void cmdConvert_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			cmdCancel.Text = "Exit";
			modVfcMain.ProcessProject();
			modDevTools.ExportSourceFiles(ref SelectedProjectGitDirectory + "\\");
		}



//UPGRADE_WARNING: Event lstVBProjects.SelectedIndexChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
		private void lstVBProjects_SelectedIndexChanged(System.Object eventSender, System.EventArgs eventArgs)
		{
			Microsoft.Vbe.Interop.VBProject objProj = default(Microsoft.Vbe.Interop.VBProject);
			short intIndex = 0;

			if (lstVBProjects.SelectedIndex > -1) {
				//UPGRADE_WARNING: Couldn't resolve default property of object Application.VBE.VBProjects. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
				objProj = AutoCADAcadApplication_definst.Application.VBE.VBProjects(this.SelectedProjectIndex);
				cmdConvert.Enabled = true;

			}

			//UPGRADE_NOTE: Object objProj may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
			objProj = null;
		}



		private void UserForm_Activate()
		{
			Microsoft.Vbe.Interop.VBProjects objProjs = default(Microsoft.Vbe.Interop.VBProjects);
			Microsoft.Vbe.Interop.VBProject objProj = default(Microsoft.Vbe.Interop.VBProject);
			short intIndex = 0;
			string strTemp = null;

			 // ERROR: Not supported in C#: OnErrorStatement


			//Load up list box with VBProjects
			//UPGRADE_WARNING: Couldn't resolve default property of object Application.VBE.VBProjects. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
			objProjs = AutoCADAcadApplication_definst.Application.VBE.VBProjects;
			cmdConvert.Enabled = false;
			//  lstMSForms.Clear
			lstVBProjects.Items.Clear();
			//UPGRADE_WARNING: Couldn't resolve default property of object lstVBProjects.ColumnCount. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
			lstVBProjects.ColumnCount = 3;
			//UPGRADE_WARNING: Couldn't resolve default property of object lstVBProjects.ColumnWidths. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
			lstVBProjects.ColumnWidths = "20 pt;90 pt;400 pt";
			short listIndex = 0;
			listIndex = 1;
			for (intIndex = 1; intIndex <= objProjs.Count; intIndex++) {
				objProj = objProjs.Item(intIndex);
				//do not list the exporter macro
				if ((objProj.fileName != objProjs.VBE.ActiveVBProject.fileName)) {
					lstVBProjects.Items.Add(Convert.ToString(intIndex));
					lstVBProjects.List(listIndex - 1, 1) = objProj.Name;
					lstVBProjects.List(listIndex - 1, 2) = objProj.fileName;
					listIndex = listIndex + 1;
				}
			}
			txtGitRepoPath.Text = Interaction.Environ("USERPROFILE") + "\\Source\\Repos\\Autocad Automation\\";
			SubExit:


			//UPGRADE_NOTE: Object objProj may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
			objProj = null;
			//UPGRADE_NOTE: Object objProjs may not be destroyed until it is garbage collected. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
			objProjs = null;
			return;
			ErrorHandler:
			switch (Err().Number) {
				case 76:
					//Path not found
					strTemp = "VFC can only work with SAVED projects," + Constants.vbCrLf + "please save all newly created projects" + Constants.vbCrLf + "and start again.";
					Interaction.MsgBox(strTemp, MsgBoxStyle.Critical, modVfcMain.ccAPPNAME);
					break;
				default:
					//Huh??
					strTemp = modVfcMain.ccAPPNAME + " ERROR" + Constants.vbCrLf + "VFC_001: Error in UserForm_Activate" + Constants.vbCrLf + "Description: " + Err().Description + Constants.vbCrLf + "Source: " + Err().Source + Constants.vbCrLf + "Number: " + Convert.ToString(Err().Number) + Constants.vbCrLf + "VFC Ver: " + modVfcMain.ccAPPVER;
					Interaction.MsgBox(strTemp, MsgBoxStyle.Critical, modVfcMain.ccAPPNAME);
					break;
			}
			this.Close();
			 // ERROR: Not supported in C#: ResumeStatement

		}

		private void UserForm_Terminate()
		{
			Debug.Print("UserForm_Terminate frmVfcMain");
		}
	}
}
